apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/plugin/PluginInitializer.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.agent.core.plugin;
+
+import org.apache.skywalking.apm.agent.core.conf.Config;
+import org.apache.skywalking.apm.agent.core.plugin.loader.AgentClassLoader;
+import org.apache.skywalking.apm.util.StringUtil;
+
+import java.util.regex.Pattern;
+
+/**
+ * create a AbstractClassEnhancePluginDefine instance according to {@link PluginDefine} and init it with agent.config
+ */
+public class PluginInitializer {
+
+    public static AbstractClassEnhancePluginDefine initialize(PluginDefine pluginDefine)
```

Comment:
```suggestion
    public static AbstractClassEnhancePluginDefine create(PluginDefine pluginDefine)
```

Reply:
You actually don't need another `create` and `init`. There is no point to split a 10-20 lines method into 3 methods, but no one is using other 2 methods.

Reply:
Could I do like this?

- rename `PluginInitializer` to `PluginCreator` and preserve create method only
- compile all rules in a static code block

![image-20210819002720789](https://gitee.com/easy4coding/blogImage/raw/master/image/image-20210819002720789.png)

Reply:
Please don't use `static code block`, instead, use a static method. `static method` is very tricky in the class loader env, let's try to avoid that.(event it should work)

Reply:
how about this?
- `loadPlugins`

![image-20210819141608769](https://gitee.com/easy4coding/blogImage/raw/master/image/image-20210819141608769.png)

- `PluginCreator#create`

Compile the rules before batch creation

![image-20210819141659466](https://gitee.com/easy4coding/blogImage/raw/master/image/image-20210819141659466.png)

Reply:
This works for me.